lang/java/avro/src/test/java/org/apache/avro/generic/TestGenericData.java

```diff
@@ -552,4 +552,28 @@ public void testToStringRecursive() throws IOException {
     }
   }
 
+  @Test
+  public void testToString() throws IOException {
```

Comment:
Since this is basically testing that identical values can be used as long as they aren't nested, I think it would be good to have this test validate that for all types. Or, have a test case for each type. That should include non-nested records that are identical. For example, a binary tree structure that has a reused leaf node should work with `toString`.